expression/column.go

```diff
@@ -223,6 +223,17 @@ func (col *CorrelatedColumn) RemapColumn(m map[int64]*Column) (Expression, error
 	}, nil
 }
 
+// HashCode implements Expression interface.
+func (col *CorrelatedColumn) HashCode(sc *stmtctx.StatementContext) []byte {
+	// Doesn't check len(hashcode) because col.Data can be changed anytime,
+	// So always use newest Datum to calc hash code.
+	col.Column.HashCode(sc)
+	if col.Data != nil {
+		col.hashcode = codec.HashCode(col.hashcode, *col.Data)
```

Comment:
don't change the original col's hash code here, after appending with another datum, the changed hash code couldn't be used as the original column's hash code anymore.

Reply:
col.hashcode actually used the original column's hashcode

Reply:
CorrelatedColumn copy original Column instead of using pointer